### ISSUE TITLE
AbstractBaseUser without is_staff field generates error

### ISSUE THREAD

**Issue by FlipperPA** [wagtail/wagtail#2777]
Pretty extreme outlier, but I ran into it. If you have an AbstractBaseUser and don't implement is_staff in the model, you run into troubles, for example, when attempting to apply Admin Permissions to a user:

```
AttributeError at /cms/users/1/
can't set attribute
Request Method: POST
Request URL:    http://vagrant.wharton.upenn.edu:8000/cms/users/1/
Django Version: 1.9.7
Exception Type: AttributeError
Exception Value:    
can't set attribute
Exception Location: /home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/wagtail/wagtailusers/forms.py in save, line 188
Python Executable:  /home/vagrant/.virtualenvs/wrds-classroom/bin/python
Python Version: 3.5.1
Python Path:    
['/vagrant/html/wrds-classroom',
 '/home/vagrant/.virtualenvs/wrds-classroom/lib/python35.zip',
 '/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5',
 '/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/plat-linux',
 '/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/lib-dynload',
 '/usr/local/lib/python3.5',
 '/usr/local/lib/python3.5/plat-linux',
 '/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages',
 '/vagrant/html/wrds-classroom']
Server time:    Thu, 23 Jun 2016 18:40:43 -0400
Traceback Switch to copy-and-paste view

/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/django/core/handlers/base.py in get_response
                    response = self.process_exception_by_middleware(e, request) ...
/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/django/core/handlers/base.py in get_response
                    response = wrapped_callback(request, *callback_args, **callback_kwargs) ...
/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/django/views/decorators/cache.py in _cache_controlled
            response = viewfunc(request, *args, **kw) ...
/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/wagtail/wagtailadmin/decorators.py in decorated_view
            return view_func(request, *args, **kwargs) ...
/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/wagtail/wagtailadmin/utils.py in wrapped_view_func
                return view_func(request, *args, **kwargs) ...
/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/wagtail/wagtailusers/views/users.py in edit
            user = form.save() ...
/home/vagrant/.virtualenvs/wrds-classroom/lib/python3.5/site-packages/wagtail/wagtailusers/forms.py in save
        user.is_staff = user.is_superuser 
```

**Comment by m1kola** [comment]
Thanks for reporting!

I'm not sure that we can rid of this. Probably we need to mention required attributes/methods somewhere in the documentation like it was done for [`django.contrib.admin`](https://docs.djangoproject.com/en/1.9/topics/auth/customizing/#custom-users-and-django-contrib-admin) (also requires `is_staff` to be implemented).

**Comment by gasman** [comment]
Setting `user.is_staff = user.is_superuser` in Wagtail's user create/edit forms is a bit of a dubious design decision anyway - see #970. It may well be that this is the only place we refer to `is_staff` in Wagtail's codebase... in which case replacing it wouldn't be a bad idea.

**Comment by FlipperPA** [comment]
I'll do some grepping through the code base to check. If that's the only place where `is_staff` is referenced, I agree!

I did have `is_staff` set as a property to return the value of `is_admin` in the `AbstractBaseUser` as well. Fortunately, I'm early enough in this process and just toying with possibilities for our custom Django user model for a net new project that I can pivot, but naturally, I want to get this right since changing a custom user model after launch is always a bit of an adventure.

None of this would be happening if Django didn't have a 30-character limit on `username`, but that's a conversation for another day. :)

**Comment by gasman** [comment]
Fixed in #3434
